pkg/registry/storage/csidriver/strategy.go

```diff
@@ -84,9 +87,16 @@ func (csiDriverStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.
 	}
 
 	// Any changes to the mutable fields increment the generation number.
-	if !apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.TokenRequests, newCSIDriver.Spec.TokenRequests) || !apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.RequiresRepublish, newCSIDriver.Spec.RequiresRepublish) {
+	if !apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.TokenRequests, newCSIDriver.Spec.TokenRequests) ||
+		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.RequiresRepublish, newCSIDriver.Spec.RequiresRepublish) ||
+		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.SELinuxMount, newCSIDriver.Spec.SELinuxMount) {
 		newCSIDriver.Generation = oldCSIDriver.Generation + 1
 	}
+
+	if oldCSIDriver.Spec.SELinuxMount == nil &&
```

Comment:
should the feature gate check go before the generation increment?

Reply:
With the alpha gate off, I think the validation allows change of SELinuxMount if it was set before, so the generation should change too.

Reply:
If the feature gate is off and somebody tries to set SELinuxMount when it wasn't set before, the strategy.go will drop the field, but generation will be incremented. Does that make sense?

Reply:
I see, it should drop the field *before* bumping the generation, as it's done for `VolumeLifecycleModes`. Moved.